### ISSUE TITLE
Store checkbox state `Show visible layers only` in the `Add Layer to Legend` dialog 

### ISSUE THREAD

**Issue by MrChebur** [qgis/QGIS#52619]
### Feature description

In `Layout`, when you click the `Add layer(s) to legend` button in the `Legend` item properties, its boolean parameter `Show visible layers only` is always reset to its default state `False` (unchecked).

It would be nice to save the state of this checkbox so you don't have to click it all the time.

![1](https://user-images.githubusercontent.com/20655102/230892642-9e0d002a-06bc-437d-85d9-d7638c472142.gif)

**Comment by bkepl** [comment]
I agree. Even better would be to have an option in the general program settings for the default behavior.
